### PR TITLE
199 show space bytes

### PR DESCRIPTION
**Pull request by tcnichol**
## Description
the statistics view for spaces now shows bytes per space. 

## Review Time Estimate
- [x] When possible

## Types of changes
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
- [x] I have updated the CHANGELOG.md.
- [x] I have signed the CLA
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: app/views/spaces/statistics.scala.html [hunk 1]
```diff
@@ -1,11 +1,37 @@
...
-        <p><span class="glyphicon glyphicon-briefcase"></span> Datasets: @space.datasetCount</p>
+    <div class="col-md-12" id="resources-panel-container">
+        <div class="panel panel-default" id="resources-panel" data-clampedwidth=".col-md-5">
+            <div class="panel-heading">
+                <h3 class="panel-title">Resources</h3>
+            </div>
+            <div class="panel-body">
+                <ul class="list-group">
+
+@*                    <li class="list-group-item" title="@humanReadableNumber(spacesCount) @Messages("spaces.title")">*@
```
Comment (robkooper) [inline app/views/spaces/statistics.scala.html:14]: if not used, please remove commented out code.

**Review by robkooper — CHANGES_REQUESTED** [review]

File: app/views/spaces/statistics.scala.html (hunk 1 above)
Comment (tcnichol) [reply app/views/spaces/statistics.scala.html]: This is now removed.

File: app/controllers/Spaces.scala [hunk 2]
```diff
@@ -648,4 +641,35 @@ class Spaces @Inject() (spaces: SpaceService, users: UserService, events: EventS
...
+  private def getBytesPerSpace(spaceId: UUID, user: models.User) : Long = {
+    val allDatasetsInSpace = datasets.listSpace(0, spaceId.toString(), Some(user))
+    var spaceBytes: Long = 0
+    for (ds <- allDatasetsInSpace){
+      val ds_files = ds.files
+      for (ds_f <- ds_files){
+        files.get(ds_f) match {
+          case Some(file) => {
+            files.getBytes(file.id) match {
+              case Some((stream, name, filetype, bytes)) => {
```
[the remark this answers is not in the document]
Comment (tcnichol) [reply app/controllers/Spaces.scala:663]: I think this is now fixed by caching the spaceBytes as part of the Space object. When datasets are added or files added, the bytes increment, or decrement if removed. So there won't be these calls anymore.

File: app/controllers/Spaces.scala [hunk 3]
```diff
@@ -648,4 +641,35 @@ class Spaces @Inject() (spaces: SpaceService, users: UserService, events: EventS
...
+    var spaceFiles: Integer = 0
+    val allDatasetsInSpace = datasets.listSpace(0, spaceId.toString(), Some(user))
+    for (ds <- allDatasetsInSpace) {
+      val files_in_ds = ds.files.length
+      spaceFiles += files_in_ds
+    }
+    spaceFiles
+  }
+
+  private def getBytesPerSpace(spaceId: UUID, user: models.User) : Long = {
```
[the remark this answers is not in the document]
Comment (max-zilla) [reply app/controllers/Spaces.scala:654]: I see where that field is implemented and changed which is good - but on the main space page you're still calling this block of code to calculate dynamically right? collections/Spaces line 656 - you should be able to check the cached property of the space instead now I think?

**Review by max-zilla — CHANGES_REQUESTED** [review]

**Comment by tcnichol** [comment]
@max-zilla will fix that and make another commit

**Comment by tcnichol** [comment]
@max-zilla I fixed the part on app/controllers. 

That should be done, I do need to test and commit the update script.

**Comment by tcnichol** [comment]
Also committed the db update for adding space bytes to existing spaces.

**Review by max-zilla — APPROVED** [review]

**Review by robkooper — CHANGES_REQUESTED** [review]
I will make some changes, this breaks if the file is "imported" and not uploaded. This also breaks in case of large databases (cursor times out).

**Review by lmarini — APPROVED** [review]

**Pull request merged**